### ISSUE TITLE
Lua Error when picking mage-type with modification "EoHS with any era"

### ISSUE THREAD

**Issue by knyghtmare** [elidupree/Era_of_High_Sorcery#173]
```ini
20210701 02:31:55 error engine: failed to auto-store $unit at (-999,-999)
20210701 02:32:01 error wml: EoHS error: [string "-- utils-units.cfg..."]:13: bad argument #1 to 'create' (WML table expected, got table)
stack traceback:
	[string "-- _initial.cfg..."]:23: in function <[string "-- _initial.cfg..."]:22>
	[C]: in field 'create'
	[string "-- utils-units.cfg..."]:13: in field 'create_wunit'
	[string "-- unit_types.cfg..."]:1535: in field 'update_all_type_adjustments'
	[string "-- initialization.cfg..."]:146: in field 'init_leader'
	[string "-- initialization.cfg..."]:174: in field 'init_settings'
	[string "-- events.cfg..."]:74: in upvalue 'func'
	[string "-- _initial.cfg..."]:41: in function <[string "-- _initial.cfg..."]:39>
	(...tail calls...)
	[C]: in function '.xpcall'
	[string "-- _initial.cfg..."]:18: in field 'capture_errors'
	[string "-- theme-item-ui.cfg..."]:108: in field 'suppress_ui_input_during'
	[string "-- _initial.cfg..."]:39: in field 'integer index'
	[string "--EoHS.set_simple_event("side_turn", ...)..."]:2: in local 'bytecode'
	lua/wml-tags.lua:265: in local 'cmd'
	lua/wml-utils.lua:144: in field 'handle_event_commands'
	lua/wml-flow.lua:5: in function <lua/wml-flow.lua:4>
20210701 02:32:01 error wml: EoHS debug: { "EoHS.begin_event notices that there was probably an error earlier; if there *wasn't* an error earlier, then *this* is an error" }
20210701 02:32:59 error wml: EoHS error: [string "-- utils-units.cfg..."]:13: bad argument #1 to 'create' (WML table expected, got table)
stack traceback:
	[string "-- _initial.cfg..."]:23: in function <[string "-- _initial.cfg..."]:22>
	[C]: in field 'create'
	[string "-- utils-units.cfg..."]:13: in field 'create_wunit'
	[string "-- unit_types.cfg..."]:1535: in field 'update_all_type_adjustments'
	[string "-- training.cfg..."]:315: in field 'train_unit_to_skillset'
	[string "-- menu.cfg..."]:463: in field '?'
	[string "-- utils-sync.cfg..."]:242: in upvalue 'func'
	[string "-- _initial.cfg..."]:41: in function <[string "-- _initial.cfg..."]:39>
	(...tail calls...)
	[C]: in function '.xpcall'
	[string "-- _initial.cfg..."]:18: in field 'capture_errors'
	[string "-- theme-item-ui.cfg..."]:108: in field 'suppress_ui_input_during'
	[string "-- _initial.cfg..."]:39: in field 'integer index'
	[string "--EoHS.set_simple_event("EoHS_become_synced",..."]:2: in local 'bytecode'
	lua/wml-tags.lua:265: in local 'cmd'
	lua/wml-utils.lua:144: in field 'handle_event_commands'
	lua/wml-flow.lua:5: in function <lua/wml-flow.lua:4>
	[C]: in field '?'
	lua/core/wml.lua:193: in function 'wml.fire'
	[string "-- utils-sync.cfg..."]:234: in field 'may_do_synced_now'
	[string "-- theme-item-ui.cfg..."]:178: in upvalue 'after_cleanup'
	[string "-- _initial.cfg..."]:82: in function <[string "-- _initial.cfg..."]:56>
20210701 02:32:59 error wml: EoHS error: EoHS.begin_event used in an existing context
stack traceback:
	[string "-- utils-misc.cfg..."]:23: in field 'err'
	[string "-- utils-misc.cfg..."]:30: in field 'assert'
	[string "-- utils-sync.cfg..."]:33: in field 'begin_event'
	[string "-- _initial.cfg..."]:40: in function <[string "-- _initial.cfg..."]:39>
	(...tail calls...)
	[C]: in function '.xpcall'
	[string "-- _initial.cfg..."]:18: in field 'capture_errors'
	[string "-- theme-item-ui.cfg..."]:108: in field 'suppress_ui_input_during'
	[string "-- _initial.cfg..."]:39: in field 'integer index'
	[string "--EoHS.set_simple_event("select", ...)..."]:2: in local 'bytecode'
	lua/wml-tags.lua:265: in local 'cmd'
	lua/wml-utils.lua:144: in field 'handle_event_commands'
	lua/wml-flow.lua:5: in function <lua/wml-flow.lua:4>
```
This was the set-up.

![Screenshot 2021-07-01 022407](https://user-images.githubusercontent.com/5283677/124027678-f801ca00-da14-11eb-88ca-397ad5311379.png)

**Comment by elidupree** [comment]
This is something I can't solve from just the error message - can you post a replay with the error?

**Comment by elidupree** [comment]
oh hang on, that's a scenario that requires download, right? and it's not on the add-on server right now? so I couldn't watch the replay anyway, sigh

**Comment by elidupree** [comment]
Update: I've just pushed a change that will make EoHS generate more-useful error messages in situations like this. Want to try again and send me the new error?

**Comment by knyghtmare** [comment]
> Update: I've just pushed a change that will make EoHS generate more-useful error messages in situations like this. Want to `git pull`, try again and send me the new error?

```ini
20210701 16:27:50 error engine: failed to auto-store $unit at (-999,-999)
20210701 16:28:09 error wml: EoHS error: EoHS.check_wml_table[variables]: WML table was not detected as invalid but still is: {
  EoHS_all_EoHS_overlays = "misc/leader-crown.png,misc/blank-hex.png~BLIT(misc/loyal-icon.png~CROP(15,8,11,4)~FL(horiz)~CHAN(0+ 120*(red)/255,0+ 255*(red)/255,0+ 0*(red)/255),46,13)~BLIT(misc/loyal-icon.png~CROP(15,8,11,4)~FL(horiz)~CHAN(0+ 120*(red)/255,0+ 255*(red)/255,0+ 0*(red)/255),46,16)~BLIT(misc/loyal-icon.png~CROP(15,8,11,4)~FL(horiz)~CHAN(0+ 120*(red)/255,0+ 255*(red)/255,0+ 0*(red)/255),46,19)~BLIT(misc/loyal-icon.png~CROP(15,8,11,4)~FL(horiz)~CHAN(0+ 120*(red)/255,0+ 255*(red)/255,0+ 0*(red)/255),46,22)~BLIT(misc/loyal-icon.png~CROP(15,8,11,4)~FL(horiz)~CHAN(0+ 120*(red)/255,0+ 255*(red)/255,0+ 0*(red)/255),46,25)~BLIT(misc/loyal-icon.png~CROP(15,8,11,4)~FL(horiz)~CHAN(0+ 120*(red)/255,0+ 255*(red)/255,0+ 0*(red)/255),46,28)~BLIT(misc/loyal-icon.png~CROP(15,8,11,4)~FL(horiz)~CHAN(0+ 120*(red)/255,0+ 255*(red)/255,0+ 0*(red)/255),46,31)~BLIT(misc/loyal-icon.png~CROP(15,8,11,4)~FL(horiz)~CHAN(0+ 120*(red)/255,0+ 255*(red)/255,0+ 0*(red)/255),46,34)~BLIT(misc/loyal-icon.png~CROP(15,8,11,4)~FL(horiz)~CHAN(0+ 120*(red)/255,0+ 255*(red)/255,0+ 0*(red)/255),46,37)~BLIT(misc/loyal-icon.png~CROP(15,8,11,4)~FL(horiz)~CHAN(0+ 120*(red)/255,0+ 255*(red)/255,0+ 0*(red)/255),46,40)~BLIT(misc/loyal-icon.png~CROP(15,8,11,4)~FL(horiz)~CHAN(0+ 120*(red)/255,0+ 255*(red)/255,0+ 0*(red)/255),46,43)",
  EoHS_best_max_moves_this_turn = 5,
  EoHS_dress_up_options_hair = "hooded",
  EoHS_dress_up_options_recolor_hair = "111,63,31",
  EoHS_dress_up_options_recolor_skin = "144,84,27",
  EoHS_ellipse = "misc/ellipse-leader-",
  EoHS_hitpoints_above_max = 0,
  EoHS_id = 1001,
  EoHS_image_mods = "~TC(1,magenta)",
  EoHS_is_starting_leader = true,
  EoHS_is_wizard = true,
  EoHS_max_hitpoints_before_change = 68,
  EoHS_moves_above_max = 0,
  EoHS_original_name = "Anonymous player 1",
  EoHS_original_type = "Custodian",
  EoHS_skillset_aegis = false,
  EoHS_skillset_altar_of_battle = false,
  EoHS_skillset_annulling_lance = false,
  EoHS_skillset_black_cat = false,
  EoHS_skillset_call_dragon = false,
  EoHS_skillset_change_terrain = false,
  EoHS_skillset_circle_of_protection = false,
  EoHS_skillset_compulsion = false,
  EoHS_skillset_construction = false,
  EoHS_skillset_corridor_of_frost = false,
  EoHS_skillset_create_portal = false,
  EoHS_skillset_curse_of_anxiety = false,
  EoHS_skillset_dark_pact = false,
  EoHS_skillset_dehydration = false,
  EoHS_skillset_dispelling_touch = false,
  EoHS_skillset_enchanter = false,
  EoHS_skillset_energy_blast = false,
  EoHS_skillset_expansive_change = false,
  EoHS_skillset_extra_apprentice_1 = false,
  EoHS_skillset_extra_apprentice_2 = false,
  EoHS_skillset_faction_Aragwaithi_EL = false,
  EoHS_skillset_faction_Dark_Legion_EL = false,
  EoHS_skillset_faction_Drakes = false,
  EoHS_skillset_faction_Dunefolk = false,
  EoHS_skillset_faction_Human_Alliance_EL = true,
  ["EoHS_skillset_faction_Knalgan Alliance"] = false,
  EoHS_skillset_faction_Minotaurs_EL = false,
  EoHS_skillset_faction_Orcs_EL = false,
  EoHS_skillset_faction_Quenoth_EL = false,
  EoHS_skillset_faction_Sylvans_EL = false,
  EoHS_skillset_faction_Undead_EL = false,
  EoHS_skillset_faction_Vampires_EL = false,
  EoHS_skillset_faction_Windsong_EL = false,
  EoHS_skillset_farseer = false,
  EoHS_skillset_fireball = false,
  EoHS_skillset_focused_study = false,
  EoHS_skillset_forceful = false,
  EoHS_skillset_greater_mentor = false,
  EoHS_skillset_group_teleport = false,
  EoHS_skillset_haste = false,
  EoHS_skillset_lightning_bolt = false,
  EoHS_skillset_mage_shield = false,
  EoHS_skillset_mentor = false,
  EoHS_skillset_metamorphosis = false,
  EoHS_skillset_meteor_storm = false,
  EoHS_skillset_mount = false,
  EoHS_skillset_mud_touch = false,
  EoHS_skillset_ogre_fury = false,
  EoHS_skillset_pall_of_fatigue = false,
  EoHS_skillset_poison_ground = false,
  EoHS_skillset_rampage = false,
  EoHS_skillset_reanimating_touch = false,
  EoHS_skillset_regeneration = false,
  EoHS_skillset_renounce_life = false,
  EoHS_skillset_repelling_gust = false,
  EoHS_skillset_restoration = false,
  EoHS_skillset_restraint = false,
  EoHS_skillset_scribe = false,
  EoHS_skillset_show_unusual = false,
  EoHS_skillset_siphon_strength = false,
  EoHS_skillset_soul_bind = false,
  EoHS_skillset_spiritform = false,
  EoHS_skillset_summon_arctic_drake = false,
  EoHS_skillset_summon_chocobone = false,
  EoHS_skillset_summon_fire_guardian = false,
  EoHS_skillset_summon_ghost = false,
  EoHS_skillset_summon_giant_scorpion = false,
  EoHS_skillset_summon_gryphon = false,
  EoHS_skillset_summon_sea_serpent = false,
  EoHS_skillset_summon_skeleton = false,
  EoHS_skillset_summon_vampire_bat = false,
  EoHS_skillset_summon_water_serpent = false,
  EoHS_skillset_summon_wolf = false,
  EoHS_skillset_summon_wose = false,
  EoHS_skillset_summon_yeti = false,
  EoHS_skillset_summoner = false,
  EoHS_skillset_swarm_nest = false,
  EoHS_skillset_undying_army = false,
  EoHS_skillset_vex_dragon_bones = false,
  EoHS_skillset_wizard_sword = false,
  EoHS_skillset_zombie_mastery = false,
  EoHS_skillset_zombie_swarm = false,
  EoHS_unit_attribute_memory_alignment = "lawful",
  EoHS_unit_attribute_memory_gender = "male",
  EoHS_unit_attribute_memory_image = "units/human-loyalists/custodian.png",
  EoHS_unit_attribute_memory_is_fearless = false,
  EoHS_unit_attribute_memory_resistance_arcane = 80,
  EoHS_unit_attribute_memory_resistance_arcane_focus = 120,
  EoHS_unit_attribute_memory_resistance_blade = 50,
  EoHS_unit_attribute_memory_resistance_cold = 100,
  EoHS_unit_attribute_memory_resistance_electric = 100,
  EoHS_unit_attribute_memory_resistance_energy = 120,
  EoHS_unit_attribute_memory_resistance_fire = 110,
  EoHS_unit_attribute_memory_resistance_impact = 90,
  EoHS_unit_attribute_memory_resistance_pierce = 60,
  EoHS_unit_attribute_memory_side = 1,
  EoHS_unit_attribute_memory_type = "Custodian",
  EoHS_unit_attribute_memory_variation = "",
  EoHS_unit_attribute_memory_x = 6,
  EoHS_unit_attribute_memory_y = 5,
  EoHS_wizard_level = 1,
  EoHS_wizard_protection = true
}
stack traceback:
	[string "-- utils-misc.cfg..."]:23: in field 'err'
	[string "-- utils-tables.cfg..."]:316: in local 'err'
	[string "-- utils-tables.cfg..."]:343: in field 'check_wml_table'
	[string "-- utils-tables.cfg..."]:323: in field 'check_wml_table'
	[string "-- unit_types.cfg..."]:1535: in field 'update_all_type_adjustments'
	[string "-- initialization.cfg..."]:146: in field 'init_leader'
	[string "-- initialization.cfg..."]:174: in field 'init_settings'
	[string "-- events.cfg..."]:74: in upvalue 'func'
	[string "-- _initial.cfg..."]:41: in function <[string "-- _initial.cfg..."]:39>
	(...tail calls...)
	[C]: in function '.xpcall'
	[string "-- _initial.cfg..."]:18: in field 'capture_errors'
	[string "-- theme-item-ui.cfg..."]:108: in field 'suppress_ui_input_during'
	[string "-- _initial.cfg..."]:39: in field 'integer index'
	[string "--EoHS.set_simple_event("side_turn", ...)..."]:2: in local 'bytecode'
	lua/wml-tags.lua:265: in local 'cmd'
	lua/wml-utils.lua:144: in field 'handle_event_commands'
	lua/wml-flow.lua:5: in function <lua/wml-flow.lua:4>
20210701 16:28:09 error wml: EoHS error: [string "-- utils-units.cfg..."]:13: bad argument #1 to 'create' (WML table expected, got table)
stack traceback:
	[string "-- _initial.cfg..."]:23: in function <[string "-- _initial.cfg..."]:22>
	[C]: in field 'create'
	[string "-- utils-units.cfg..."]:13: in field 'create_wunit'
	[string "-- unit_types.cfg..."]:1536: in field 'update_all_type_adjustments'
	[string "-- initialization.cfg..."]:146: in field 'init_leader'
	[string "-- initialization.cfg..."]:174: in field 'init_settings'
	[string "-- events.cfg..."]:74: in upvalue 'func'
	[string "-- _initial.cfg..."]:41: in function <[string "-- _initial.cfg..."]:39>
	(...tail calls...)
	[C]: in function '.xpcall'
	[string "-- _initial.cfg..."]:18: in field 'capture_errors'
	[string "-- theme-item-ui.cfg..."]:108: in field 'suppress_ui_input_during'
	[string "-- _initial.cfg..."]:39: in field 'integer index'
	[string "--EoHS.set_simple_event("side_turn", ...)..."]:2: in local 'bytecode'
	lua/wml-tags.lua:265: in local 'cmd'
	lua/wml-utils.lua:144: in field 'handle_event_commands'
	lua/wml-flow.lua:5: in function <lua/wml-flow.lua:4>
20210701 16:28:09 error wml: EoHS debug: { "EoHS.begin_event notices that there was probably an error earlier; if there *wasn't* an error earlier, then *this* is an error" }
```

**Comment by elidupree** [comment]
fixed in bcec88403688508a19d5b7584045fd906a29affb
